### ISSUE TITLE
Enforce `.replaceAll()` over `.replace()` when the pattern has `g` flag

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1948]
I believe they are the same, but use `.replaceAll()` has more clear intention

# Fail

```js
foo.replace(/\s/g, '')
```

# Pass

```js
foo.replaceAll(/\s/g, '')
```

```js
foo.replace(/\s/, '')
```

```js
foo.replace(" ", '')
```

**Comment by sindresorhus** [comment]
Accepted.

**Comment by kachkaev** [comment]
Do both `foo.replaceAll(/\s/g, '')` and `foo.replaceAll(/\s/, '')` pass? Feels like there should be one style or another everywhere.

**Comment by fisker** [comment]
`foo.replaceAll(/\s/, '')` is a runtime error. https://github.com/ota-meshi/eslint-plugin-regexp/issues/472
